### DISCUSSION TITLE
UR5 Accurate Kinematics through DH parameters

### DISCUSSION THREAD

**Discussion by jirrivdb** [visose/Robots#172] [Q&A] [answered]
Hi All,

I am currently looking into setting up our UR5 robot with its own specific denavit-hartenberg parameters. I know that a robot XML can be edited in which the a(mm) and b(mm) values can be changed. UR packages these values in a URP exported from the robot. I have previously gone through the same process in roboDK in which the DH parameters are extracted from the URP file and the accuracy of the simulation is increased. The values loaded by RoboDK and present in the URP are the following: 
![RoboDK_UR5_DH Table](https://user-images.githubusercontent.com/74723126/181042329-d36fc569-71f9-4f33-af80-6655ee1dee9d.png)

The standard values for the UR5 seem to be the following:
![Standard_UR5_DH Table](https://user-images.githubusercontent.com/74723126/181042837-21ce0f3f-3a99-48ee-94b9-63e720c44cdd.png)

These values are very different from each other. In roboDK this does result in a more accurate simulation. In the Robots plugin I have not been able to reach the same result. I think this is due to not being able to change the alpha and theta parameters.
So the question is: Is there any way to change the alpha and theta values or can this problem be approached from another direction?

**Comment by visose** [comment, marked as answer]
Each robot type has a start pose (a list of joint angles) that unfortunately is not exposed as a variable that can be changed by the user. Currently this is per robot manufacturer (UR, ABB, etc...).

But in any case, these simulation errors are due to the small defects in the physical robot compared to the ideal robot. In a small robot like a UR5, this error would be negated when calibrating a reference frame in your environment.

If you are noticing this error, my guess is you are not calibrating a reference frame. This is usually done by jogging the robot to a few points in space (at least 3) and using the values shown in the pendant to create a plane. Then create your targets relative to this plane (using a Frame component).

This will offset both the error due to the robot's deviation from ideal values but also your environment errror which is usually larger.

**Comment by jirrivdb** [comment]
Thank you for the quick reply, this should solve my issue in the future.
